api/tests/integration/ core/test_commands.py

```diff
@@ -0,0 +1,138 @@
+import io
+import sys
+
+import pytest
+from django.core.management import call_command
+from django.db.models import Model
+from pytest_django.fixtures import SettingsWrapper
+from pytest_lazyfixture import lazy_fixture
+
+from organisations.models import (
+    Organisation,
+    OrganisationRole,
+    UserOrganisation,
+)
+from projects.models import Project
+from users.models import FFAdminUser
+
+pytestmark = pytest.mark.django_db
+
+
+def _assert_bootstrapped(
+    expected_user_email: str,
+    expected_user_password: str,
+    expected_organisation_name: str,
+    expected_project_name: str,
+) -> None:
+    assert FFAdminUser.objects.count() == 1
+    assert Organisation.objects.count() == 1
+    assert Project.objects.count() == 1
+
+    user = FFAdminUser.objects.first()
+    assert user.email == expected_user_email
+    assert user.is_superuser
+    assert user.check_password(expected_user_password)
+    organisation = Organisation.objects.first()
+    assert organisation.name == expected_organisation_name
+    user_organisation = UserOrganisation.objects.first()
+    assert user_organisation.user == user
+    assert user_organisation.organisation == organisation
+    assert user_organisation.role == OrganisationRole.ADMIN
+    project = Project.objects.first()
+    assert project.name == expected_project_name
+    assert project.organisation == organisation
+
+
+def test_bootstrap__empty_instance__creates_expected(settings: SettingsWrapper) -> None:
+    # Given
+    expected_user_email = "test@example.com"
+    expected_user_password = "foobar"
+    expected_organisation_name = "Test Org"
+    expected_project_name = "Test Project"
+
+    settings.ADMIN_EMAIL = expected_user_email
+    settings.ADMIN_INITIAL_PASSWORD = expected_user_password
+    settings.ORGANISATION_NAME = expected_organisation_name
+    settings.PROJECT_NAME = expected_project_name
+
+    # When
+    call_command("bootstrap")
+
+    # Then
+    _assert_bootstrapped(
+        expected_user_email=expected_user_email,
+        expected_user_password=expected_user_password,
+        expected_organisation_name=expected_organisation_name,
+        expected_project_name=expected_project_name,
+    )
+
+
+def test_bootstrap__empty_instance__cli_overrides__creates_expected(
+    monkeypatch: pytest.MonkeyPatch,
+) -> None:
+    # Given
+    expected_user_email = "test@example.com"
+    expected_user_password = "foobar"
+    expected_organisation_name = "Test Org"
+    expected_project_name = "Test Project"
+
+    # simulate pipe
+    monkeypatch.setattr(sys, "stdin", io.StringIO(expected_user_password + "\n"))
```

Comment:
This is cool.

Reply:
I don't quite like monkeypatching (and ORM calls) in my integration tests... but thanks!